nest/src/command/install.rs

```diff
@@ -1,123 +1,84 @@
-                                eprintln!("{}: found {} packages with name \"{}\". Please be more explicit.", red!("error"), len, purple!(target));
-                                for (i, package) in manifests.iter().enumerate() {
-                                    eprintln!("\t- {}", package);
-                                    if i == 9 && len > 10 {
-                                        eprintln!("and {} more...", len - 10);
-                                        break;
-                                    }
-                                }
-                                process::exit(1);
-                            }
-                        }
-                    }
-                    Err(e) => {
-                        eprintln!("{}: couldn't go through cache: {}.", red!("error"), e);
-                        process::exit(1);
-                    }
-                }
+    let r = sys.installer(&target.data_path(config), &target.manifest())
+        .perform(|state, progression| {
+            // Update the action only when it's significant
+            if old_state != state {
+                pb.set_action(state.to_string());
+                old_state = state;
             }
-            None => {
-                eprintln!(
-                    "{}: target \"{}\" isn't a valid target name.",
-                    red!("error"),
-                    purple!(target),
-                );
-                process::exit(1);
+            if let Some((cur, max)) = progression {
+                pb.set_max(max);
+                pb.update(cur);
+            } else {
+                pb.set_max(0);
+                pb.update(0);
             }
-        }
-    }
-
-    // Download and install packages
-    let len = targets.len();
-    for (i, target) in targets.iter().enumerate() {
-        let repo = target.repository();
-        for mirror in repo.mirrors() {
-            let target_path = target.data_path(config);
-
-            if target_path
-                .parent()
-                .map(|x| fs::create_dir_all(x).ok())
-                .is_some()
-            {
-                let mut pb = ProgressBar::new(String::from("download"));
-                pb.set_target(format!(
-                    "({}/{}) {}",
-                    i + 1,
-                    len,
-                    target.manifest().metadatas().name()
-                ));
+        });
+    pb.finish(&r);
+    r
+}
 
-                let r = repo.download(
-                    config,
-                    mirror,
-                    target.manifest(),
-                    &target_path,
-                    |cur: f64, max: f64| {
-                        pb.set_max(max as usize);
-                        pb.update(cur as usize);
-                        true
-                    },
-                );
+/// Installs all the given packages
+///
+/// This will go through all targets, checks that they exist, resolves the dependency graph,
```

Comment:
*This will go through all targets, check~~s~~ that they exist, resolve~~s~~ the dependency graph,* etc. (until the end of the sentence)

Reply:
To be clear, the sentence should at the third person.

Reply:
(Took me a couple of seconds to understand what you meant, until i saw your 2nd comment :D)